src/main/java/org/opensearch/security/setting/DeprecatedSettings.java

```diff
@@ -28,4 +28,16 @@ public static void checkForDeprecatedSetting(final Settings settings, final Stri
             );
         }
     }
+
+    /**
+     * Logs that a specific setting is deprecated, including a specific supplemental message parameter containing information that details where this setting can be removed from. Should be used in cases where a setting is not supported by the codebase and processing it would introduce errors on setup.
+     */
+    public static void logCustomDeprecationMessage(final String deprecationLocationInformation, final String deprecatedSettingKey) {
+        DEPRECATION_LOGGER.deprecate(
+            deprecatedSettingKey,
+            "In OpenSearch v2.0.0+ the setting '{}' is deprecated, it should be removed from the relevant config file using the following location information: "
```

Comment:
```suggestion
            "The setting '{}' is deprecated, it should be removed from the relevant config file using the following location information: "
```

Do you think we need to include a version in this log message? If the version is needed here can it be obtained using the [Version](https://github.com/opensearch-project/OpenSearch/blob/main/libs/core/src/main/java/org/opensearch/Version.java) class from core?

Reply:
Not sure if it's necessary I was going with the requested changes from before, should version be used?

Reply:
I think giving more information might be better so I'll leave it in for now, I'll use the version class to keep it up to date

Reply:
Nit; I'd drop the version number completely, the version when a setting was deprecated != the current version.  However, this is pretty minor IMO